harness/determined/profiler.py

```diff
@@ -488,6 +577,98 @@ def run(self) -> None:
                 batch_start_time = time.time()
 
 
+class TimingsBatcherThread(threading.Thread):
+    """
+    This is a thread that exists solely so that we can batch Timings and ship them to the
+    SenderThread every FLUSH_INTERVAL seconds.
+    """
+
+    FLUSH_INTERVAL = 10  # How often to make API calls
+
+    def __init__(
+        self,
+        trial_id: str,
+        agent_id: str,
+        inbound_queue: queue.Queue,
+        send_queue: queue.Queue,
+    ) -> None:
+        self.inbound_queue = inbound_queue
+        self.send_queue = send_queue
+        self.current_batch = TimingsBatcher(trial_id, agent_id)
```

Comment:
It seems like `current_batch` should be a local variable inside of `run()`.

Also, I can't find this `TimingsBatcher` anywhere, where is it defined?

Reply:
It's on like 821